### ISSUE TITLE
[Non-text content and media] Non-text content doesn't have a text alternative. (10.01.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#59462]
#### VFS team name
MHV

#### VFS product name
My HealtheVet

#### VFS feature name
Secure Messaging

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Non-text content doesn't have a text alternative.

### Issue Details
When viewing a list of messages ([/inbox/](https://staging.va.gov/my-health/secure-messages/inbox), [/sent/](https://staging.va.gov/my-health/secure-messages/sent), [/drafts/](https://staging.va.gov/my-health/secure-messages/drafts), or in collapsed accordion messages in a conversation), messages with attachments have a paperclip icon to indicate that an attachment is present. That icon doesn't have alternative text, or any alternative way of communicating that an attachment is present (eg. extra `aria-label` on another element).

### Link, screenshot or steps to recreate
Example message markup:

```
<div class="thread-list-item vads-l-row vads-u-padding-y--1p5 vads-u-border-bottom--1px vads-u-border-color--gray-light" data-testid="thread-list-item">
   <div class="unread-column vads-l-col"></div>
   <div class="vads-l-col vads-u-margin-left--1">
      <div class="vads-u-margin-y--0p5">
         <div>
            <div>To: MELVIN FREEMAN OERR TG_DAYT29 (Team: MELVIN FREEMAN OERR TG_DAYT29) </div> 
         </div>
       </div>
       <a class="message-subject-link vads-u-margin-y--0p5" href="/my-health/secure-messages/thread/2831943">General: Health test</a>
       <p class="received-date vads-u-margin-y--0p5">
          <i class="vads-u-margin-right--1 vads-u-font-size--sm fas fa-paperclip" aria-hidden="true"></i>
          <span>May 31, 2023 at 9:22 a.m. CDT</span>
       </p>
   </div>
</div>
```

Note that the paperclip icon `<i>` has no text equivalent indicating that there's an attachment.

### VA.gov Experience Standard
[Category Number 10, Issue Number 01](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 1.1.1_A

### Platform Recommendation
There are several ways you could add some kind of text alternative, but I think what I'd recommend here is inserting a `<span class="sr-only">Has attachment</span>` after the `<i>` icon. I would also recommend associating that text alternative with the message subject link using `aria-describedby`, so that a screen reader user who is tabbing through all the message links hears that the message has an attachment without having to explore the context around the message.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by bryan-riley-va** [comment]
JIRA ticket created: https://jira.devops.va.gov/browse/MHV-45986

**Comment by vsaleem** [comment]
Hi @shiragoodman, the Pull request for this concern is ready:[ ](https://github.com/department-of-veterans-affairs/vets-website/pull/24514)

The requirements above instructed that we describe the paperclip `<icon>` aria-label, as "Has attachments". This may sound confusing to other users... Can you confirm the wording or should we use a different text label, "e.g., Attachment icon"?

**Comment by vsaleem** [comment]
https://github.com/department-of-veterans-affairs/vets-website/pull/24514

**Comment by AngelaFowler82** [comment]
I can now tell when I arrow through the message list what messages have attachments, but not when I tab through, as I am only hearing link text. Don't forget to associate the span class with the link using ARIA-describedby.

**Comment by AngelaFowler82** [comment]
I can tell messages are unread by arrowing down the page and tabbing through the links. Closing this as resolved.

**Comment by AngelaFowler82** [comment]
I closed the wrong ticket.

**Comment by vsaleem** [comment]
> I can now tell when I arrow through the message list what messages have attachments, but not when I tab through, as I am only hearing link text. Don't forget to associate the span class with the link using ARIA-describedby.

I will take another look at this.

**Comment by vsaleem** [comment]
> I can now tell when I arrow through the message list what messages have attachments, but not when I tab through, as I am only hearing link text. Don't forget to associate the span class with the link using ARIA-describedby.

For this update, the user should hear when a link text contains an attachment whenever you tab through.  You should hear the name of the link + "Has attachment."

**Comment by vsaleem** [comment]
> Have you pushed the latest update to staging? As you can see, there is no ARIA-describedby associating the link with the Span which contains "message has attachment."

Hi @AngelaFowler82. I pushed the latest changes.  The updates are in staging.

**Comment by AngelaFowler82** [comment]
Tabbing through links, my screen reader now tells me if a message has attachments. Closing as resolved.
